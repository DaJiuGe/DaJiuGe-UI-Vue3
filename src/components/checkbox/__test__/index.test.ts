import { describe, expect, it } from 'vitest'
import { flushPromises, mount, shallowMount } from '@vue/test-utils'
import SuCheckbox from '../index.vue'

describe('Checkbox', () => {
  it('should toggle checked when click', async () => {
    const wrapper = mount(SuCheckbox, {
      attachTo: document.body,
    })
    await wrapper.find('input').trigger('click')
    expect(wrapper.find('.su-checkbox').exists()).toBeTruthy()
    await wrapper.trigger('click')
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.su-checkbox.is-checked').exists()).toBeTruthy()
  })
})
