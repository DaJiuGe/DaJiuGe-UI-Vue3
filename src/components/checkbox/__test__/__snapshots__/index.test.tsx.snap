// Vitest Snapshot v1

exports[`Checkbox > should toggle checked when click 1`] = `"<label class=\\"su-checkbox\\" data-v-5ac25baf=\\"\\"><span class=\\"su-checkbox__input\\" data-v-5ac25baf=\\"\\"><span class=\\"su-checkbox__inner\\" data-v-5ac25baf=\\"\\"></span><input type=\\"checkbox\\" class=\\"su-checkbox__original\\" data-v-5ac25baf=\\"\\"></span><span class=\\"su-checkbox__label\\" data-v-5ac25baf=\\"\\"></span></label>"`;

exports[`Checkbox > should toggle checked when click 2 1`] = `"<label class=\\"su-checkbox\\" data-v-5ac25baf=\\"\\"><span class=\\"su-checkbox__input\\" data-v-5ac25baf=\\"\\"><span class=\\"su-checkbox__inner\\" data-v-5ac25baf=\\"\\"></span><input type=\\"checkbox\\" class=\\"su-checkbox__original\\" data-v-5ac25baf=\\"\\"></span><span class=\\"su-checkbox__label\\" data-v-5ac25baf=\\"\\"></span></label>"`;
